core/runtime.rs

```diff
@@ -1391,17 +1382,15 @@ impl JsRuntime {
     exception_to_err_result(scope, exception, true)
   }
 
-  // Respond using shared queue and optionally overflown response
+  // Send finished responses to JS
   fn async_op_response(
     &mut self,
-    overflown_responses: Vec<(OpId, Box<[u8]>)>,
+    async_responses: Vec<(PromiseId, OpResponse)>,
   ) -> Result<(), AnyError> {
     let state_rc = Self::state(self.v8_isolate());
 
-    let shared_queue_size = state_rc.borrow().shared.size();
-    let overflown_responses_size = overflown_responses.len();
-
-    if shared_queue_size == 0 && overflown_responses_size == 0 {
+    let async_responses_size = async_responses.len();
+    if async_responses_size == 0 {
       return Ok(());
     }
 
```

Comment:
Side note: this conditional could potentially be removed after this PR - please see `FIXME` below

Reply:
Yes, there was a specific condition to trigger the problem and it was because of shared queue, but since shared queue is gone it shouldn't happen anymore.